### ISSUE TITLE
Missing coverage: "next" only accessed once for iterator in IterableToList

### ISSUE THREAD

**Issue by anba** [tc39/test262#1362]
We currently don't have coverage for tc39/ecma262#988 for the iterator in IterableToList.

Found while working on https://bugzilla.mozilla.org/show_bug.cgi?id=1407588.

**Comment by littledan** [comment]
I don't think we have much coverage of tc39/ecma262#988 for for-of loops or almost anything else using the iteration protocol, for that matter. The change has not yet landed in V8, but when updating test262 to the latest version, the only tests I see updated are for the async-from-sync iterator, updated in 1144b5a86b720ddff41c1f2169b5d2c216cb992f (thanks, @caitp!).

cc @kmiller68

**Comment by caitp** [comment]
There's https://github.com/tc39/test262/commit/f3b5a1e4c06aea72ddfa9c90d04f3dcb726081df which includes an update to a for-of test, and https://github.com/tc39/test262/commit/0881a6a2ad2883329d64c1c01ca294a7dcc57af7 which includes an update to an async yield* tests, as well as the async-from-sync tests.

But, it seems like there isn't the same style of coverage for ES6 `yield*`, or for-await-of tests, or spread operations, etc

**Comment by littledan** [comment]
My mistake; these tests were checked in earlier than the other ones, and I didn't notice that they were already marked as failing in V8.
